java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSink.java

```diff
@@ -167,168 +127,250 @@ private static List<String> getPkColumnNames(
         return pkColumnNames;
     }
 
-    private PreparedStatement prepareInsertStatement(SinkRow row) {
-        if (row.getOp() != Data.Op.INSERT) {
-            throw Status.FAILED_PRECONDITION
-                    .withDescription("unexpected op type: " + row.getOp())
-                    .asRuntimeException();
-        }
-        try {
-            var preparedStmt = insertPreparedStmt;
-            jdbcDialect.bindInsertIntoStatement(preparedStmt, conn, getTableSchema(), row);
-            preparedStmt.addBatch();
-            return preparedStmt;
-        } catch (SQLException e) {
-            throw io.grpc.Status.INTERNAL
-                    .withDescription(
-                            String.format(ERROR_REPORT_TEMPLATE, e.getSQLState(), e.getMessage()))
-                    .withCause(e)
-                    .asRuntimeException();
-        }
-    }
+    @Override
+    public void write(Iterable<SinkRow> rows) {
+        final int maxRetryCount = 4;
+        int retryCount = 0;
+        while (true) {
+            try {
+                // fill prepare statements with parameters
+                for (SinkRow row : rows) {
+                    if (row.getOp() == Data.Op.UPDATE_DELETE) {
+                        updateFlag = true;
```

Comment:
Do we need to record the value of `updateFlag` before entring the for loop and reset it when retry happens?